### PR TITLE
feat: Add support for overriding kratos deployment entrypoint (to allow sourcing DB credentials dynamically)

### PR DESCRIPTION
**Pull request by eh-steve**
Update kratos helm chart to allow overriding default entrypoint to support exporting environment variables using dynamic secrets.

The immediate use case is for getting a dynamic DB connection string issued by Hashicorp Vault, instead of a static kubernetes secret. 

## Checklist

- [ ] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md)
      and signed the CLA.
- [ ] I have referenced an issue containing the design document if my change
      introduces a new feature.
- [ ] I have read the [security policy](../security/policy).
- [ ] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security vulnerability, I
      confirm that I got green light (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [ ] I have added tests that prove my fix is effective or that my feature
      works.
- [ ] I have added necessary documentation within the code base (if
      appropriate).

### CODE REVIEW

File: helm/charts/kratos/templates/deployment.yaml [hunk 1]
```diff
@@ -93,8 +93,15 @@ spec:
           image: {{ include "kratos.image" . }}
           imagePullPolicy: {{ include "kratos.imagePullPolicy" . }}
           command:
+          {{ if not (empty .Values.deployment.overrideArgs.command) }}
```
Comment (Demonsthere) [inline helm/charts/kratos/templates/deployment.yaml:96]: The check is redundant, you should be able to use just `if .Values.deployment.overrideArgs.command`

File: helm/charts/kratos/values.yaml [hunk 2]
```diff
@@ -185,6 +185,13 @@ deployment:
   # - --sqa-opt-out
   extraArgs: []
 
+  # Set these to override the entrypoint of the deployment container
+  # (e.g. to source dynamic secrets or export environment dynamic variables)
+  overrideArgs:
+    command: []
```
Comment (Demonsthere) [inline helm/charts/kratos/values.yaml:191]: Not sure if an array is a good idea here. For more complex scenarios, passing a multiline string like 
```yaml
command: |
  - some very complex
  - stuff that requires {{ .Values.other.stuff }}
```

and then calling `toYaml | tpl` might be better

**Review by Demonsthere — CHANGES_REQUESTED** [review]
Hi there, thanks for the PR!

I see the benefit of such an option, but have some remarks. 
- For consistency, I'd prefer to include it in all charts, and not only one
- In charts that have several deployment objects (deployment/statefulset/jobs) it would be smart to have it per type
- I think a better naming convention would be `customComand` and `customArgs`

File: helm/charts/kratos/values.yaml (hunk 2 above)
Comment (Demonsthere) [reply helm/charts/kratos/values.yaml]: Please refer to the [Kartos chart](https://github.com/ory/k8s/blob/master/helm/charts/kratos/templates/deployment.yaml#L47-L51) how we handle similar parsing issues and how to [use it in values](https://github.com/ory/k8s/blob/master/.circleci/values/kratos.yaml#L83-L92)

**Comment by adamstrawson** [comment]
Has there been any further progress on this? Looking to include the same behaviour in some other charts, and waiting for this to be completed so I can follow the same method for consistency.

**Comment by eh-steve** [comment]
Sorry about that, we ended up implementing a few more changes in another working branch but forgot to bring them into this one - will update

**Comment by Demonsthere** [comment]
Closing due to https://github.com/ory/k8s/pull/504 implementing similar feature set :)

**Pull request closed without merging**